src/libhirte/common/cfg.c

```diff
@@ -135,8 +137,8 @@ int cfg_load_from_file(struct config *config, const char *config_file) {
         if (config_file == NULL) {
                 return -EINVAL;
         }
-        if (access(config_file, F_OK) != 0) {
-                return -ENOENT;
+        if (access(config_file, R_OK) != 0) {
```

Comment:
`R_OK` performs an existence check as well as a read check. Why encounter a possible read-access error later, when we can check earlier? Also, according to the [man page](https://man7.org/linux/man-pages/man2/access.2.html), access can also return other errors which we would mask the actual error. But since its including `ENOENT`, the above logic is not compromised.

Reply:
Yes, but making the file not readable is the similar error as invalid content inside it, it's clearly user error.  But OK